mixed/spark/common/src/main/java/com/netease/arctic/spark/SessionCatalogBase.java

```diff
@@ -0,0 +1,242 @@
+    if (isManagedSubTable(ident)) {
+      // if it's a sub table identifier, must be a managed table
+      return getTargetCatalog().loadTable(ident);
+    }
+
+    Table table = getSessionCatalog().loadTable(ident);
+    if (isManagedTable(table)) {
+      return getTargetCatalog().loadTable(ident);
+    }
+    return table;
+  }
+
+  protected boolean isManagedSubTable(Identifier ident) {
+    return false;
+  }
+
+  /** Check if the provider is managed by target catalog. */
+  protected abstract boolean isManagedProvider(String provider);
+
+  @Override
+  public Table createTable(
+      Identifier ident, StructType schema, Transform[] partitions, Map<String, String> properties)
+      throws TableAlreadyExistsException, NoSuchNamespaceException {
+    String provider = properties.get("provider");
+    if (isManagedProvider(provider)) {
+      return getTargetCatalog().createTable(ident, schema, partitions, properties);
+    } else {
+      // delegate to the session catalog
+      return getSessionCatalog().createTable(ident, schema, partitions, properties);
+    }
+  }
+
+  @Override
+  public Table alterTable(Identifier ident, TableChange... changes) throws NoSuchTableException {
+    Table table = getSessionCatalog().loadTable(ident);
+    if (isManagedTable(table)) {
+      return getTargetCatalog().alterTable(ident, changes);
+    } else {
+      return getSessionCatalog().alterTable(ident, changes);
+    }
+  }
+
+  @Override
+  public boolean dropTable(Identifier ident) {
+    // no need to check table existence to determine which catalog to use. if a table doesn't exist
+    // then both are
+    // required to return false.
+    try {
+      Table table = getSessionCatalog().loadTable(ident);
+      if (isManagedTable(table)) {
+        return getTargetCatalog().dropTable(ident) || getSessionCatalog().dropTable(ident);
+      } else {
+        return getSessionCatalog().dropTable(ident);
+      }
+    } catch (NoSuchTableException e) {
+      return getSessionCatalog().dropTable(ident);
+    }
+  }
+
+  @Override
+  public boolean purgeTable(Identifier ident) throws UnsupportedOperationException {
+    // no need to check table existence to determine which catalog to use. if a table doesn't exist
+    // then both are
+    // required to return false.
+    try {
+      Table table = getSessionCatalog().loadTable(ident);
+      if (isManagedTable(table)) {
+        return getTargetCatalog().purgeTable(ident) || getSessionCatalog().purgeTable(ident);
+      } else {
+        return getSessionCatalog().purgeTable(ident);
+      }
+    } catch (NoSuchTableException e) {
+      return getSessionCatalog().purgeTable(ident);
+    }
+  }
+
+  @Override
+  public void renameTable(Identifier from, Identifier to)
+      throws NoSuchTableException, TableAlreadyExistsException {
+    // rename is not supported by HadoopCatalog. to avoid UnsupportedOperationException for session
```

Comment:
The comments here seem to have been broken due to automatic formatting.

Reply:
fixed